pandas/core/internals/construction.py

```diff
@@ -559,11 +559,12 @@ def sanitize_array(data, index, dtype=None, copy=False,
 
             # possibility of nan -> garbage
             if is_float_dtype(data.dtype) and is_integer_dtype(dtype):
```

Comment:
I had to change this becuase ``is_integer_dtype`` picks up nullable-integer types, I think it was just buggy before.

Reply:
does this need a IntNA test then?

Reply:
I think this is my only unresolved comment.